frontend/ui/src/pages/Channels/ChannelsMainPage/index.tsx

```diff
@@ -29,8 +56,14 @@ const ChannelsMainPage = (props: ChannelsConnectProps) => {
       </div>
 
       <div className={styles.wrapper}>
+        {displayDialogFromSource !== '' && openRequirementsDialog(displayDialogFromSource)}
         {props.config.components['sources-chatplugin'].enabled && <ChatPluginSource pluginSource={props.channels} />}
-        {props.config.components['sources-facebook'].enabled && <FacebookSource facebookSource={props.channels} />}
+        {props.config.components['sources-facebook'].enabled && (
+          <FacebookSource
+            facebookSource={props.channels}
+            showDialogAction={(source: string) => setDisplayDialogFromSource(source)}
+          />
+        )}
```

Comment:
this code reads always the same no matter the source, that's a hint we can simplify the code a bit (also because the naming here is off... why are we passing channels to something called `xSource`?).

One way of doing this creating an helper function that takes the component name and the relevant props and switches on the right component. Then we can `props.config.components.map((name, component) => component.enabled && getSource(name, props))`

Reply:
I think we should pass the channels already filtered to the component, this way it would be way more cleaner. But maybe we can do this in a different PR, I don't want to be too annoying. If it's not fixed in this PR, please create the issue before merging.